### PR TITLE
Match multi-line mocks and support default imports

### LINKED ISSUE #2: Multi-line mocks are not recognized
1. Create test with a multi-line mock (e.g. below)
2. Run `jest_lint`

Expected: Success

Actual:
```
  Missing mocks:
    jest.mock("react-dom")
```

Test Code:
```
jest.mock("react-dom", () => ({
  ...jest.requireActual<typeof ReactDOM>("react-dom"),
  useFormStatus: jest.fn(),
}));
```

### PR DESCRIPTION
**Pull request by aquacash5**
Fixes #2 

Fixes:
- Move `TestPair` and `Module` into new files.
- Fix matching multi-line mocks.
- Add matching `*` and default imports.
- Move Regex to single initialization instead of compiling the regex for each file.

### CODE REVIEW

**Comment by nbolton** [comment]
@OwenMPhillips Would you mind testing this since you're in Synergy 3 land at the moment? If it works for you please approve, I'll go ahead and review the code.

**Review by OwenMPhillips — CHANGES_REQUESTED** [review]
Running `cargo run -- --help` gives me these errors:

```
error[E0658]: use of unstable library feature 'lazy_cell'
  --> src/main.rs:10:27
   |
10 | use std::{fs, path::Path, sync::LazyLock};
   |                           ^^^^^^^^^^^^^^
   |
   = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information

error[E0658]: use of unstable library feature 'lazy_cell'
  --> src/main.rs:16:22
   |
16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
   |                      ^^^^^^^^^^^^^^^
   |
   = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information

error[E0658]: use of unstable library feature 'lazy_cell'
  --> src/main.rs:16:40
   |
16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
   |                                        ^^^^^^^^
   |
   = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information

error[E0658]: use of unstable library feature 'lazy_cell'
  --> src/main.rs:19:29
   |
19 | static IGNORE_IMPORT_REGEX: LazyLock<Regex> =
   |                             ^^^^^^^^^^^^^^^
   |
   = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information

error[E0658]: use of unstable library feature 'lazy_cell'
  --> src/main.rs:20:5
   |
20 |     LazyLock::new(|| Regex::new(r"//#region not-mocked.*?//#endregion").unwrap());
   |     ^^^^^^^^
   |
   = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information

error[E0658]: use of unstable library feature 'lazy_cell'
  --> src/main.rs:16:40
   |
16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
   |                                        ^^^^^^^^^^^^^
   |
   = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information

error[E0658]: use of unstable library feature 'lazy_cell'
  --> src/main.rs:20:5
   |
20 |     LazyLock::new(|| Regex::new(r"//#region not-mocked.*?//#endregion").unwrap());
   |     ^^^^^^^^^^^^^
   |
   = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `jest_lint` (bin "jest_lint") due to 7 previous errors
```

**Comment by OwenMPhillips** [comment]
> Running `cargo run -- --help` gives me these errors:
> 
> ```
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:10:27
>    |
> 10 | use std::{fs, path::Path, sync::LazyLock};
>    |                           ^^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:16:22
>    |
> 16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
>    |                      ^^^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:16:40
>    |
> 16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
>    |                                        ^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:19:29
>    |
> 19 | static IGNORE_IMPORT_REGEX: LazyLock<Regex> =
>    |                             ^^^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:20:5
>    |
> 20 |     LazyLock::new(|| Regex::new(r"//#region not-mocked.*?//#endregion").unwrap());
>    |     ^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:16:40
>    |
> 16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
>    |                                        ^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:20:5
>    |
> 20 |     LazyLock::new(|| Regex::new(r"//#region not-mocked.*?//#endregion").unwrap());
>    |     ^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> For more information about this error, try `rustc --explain E0658`.
> error: could not compile `jest_lint` (bin "jest_lint") due to 7 previous errors
> ```

Whoopse. Needed to update Rust.

**Comment by OwenMPhillips** [comment]
> Running `cargo run -- --help` gives me these errors:
> 
> ```
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:10:27
>    |
> 10 | use std::{fs, path::Path, sync::LazyLock};
>    |                           ^^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:16:22
>    |
> 16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
>    |                      ^^^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:16:40
>    |
> 16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
>    |                                        ^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:19:29
>    |
> 19 | static IGNORE_IMPORT_REGEX: LazyLock<Regex> =
>    |                             ^^^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:20:5
>    |
> 20 |     LazyLock::new(|| Regex::new(r"//#region not-mocked.*?//#endregion").unwrap());
>    |     ^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:16:40
>    |
> 16 | static IMPORT_REGEX: LazyLock<Regex> = LazyLock::new(|| {
>    |                                        ^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> error[E0658]: use of unstable library feature 'lazy_cell'
>   --> src/main.rs:20:5
>    |
> 20 |     LazyLock::new(|| Regex::new(r"//#region not-mocked.*?//#endregion").unwrap());
>    |     ^^^^^^^^^^^^^
>    |
>    = note: see issue #109736 <https://github.com/rust-lang/rust/issues/109736> for more information
> 
> For more information about this error, try `rustc --explain E0658`.
> error: could not compile `jest_lint` (bin "jest_lint") due to 7 previous errors
> ```

Just needed to update Rust.

**Review by OwenMPhillips — APPROVED** [review]

**Review by OwenMPhillips — APPROVED** [review]

File: src/test_pair.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            .replace(SPEC_FILE_EXT, "");
+        let under_test_file = test_file.with_file_name(name);
+        if !(under_test_file.exists()) {
+            return Err(Error::UnderTestFileDoesNotExist(under_test_file));
+        }
+        Ok(Self::new(test_file, under_test_file))
+    }
+}
+
+fn find_all_tests_in_directory_internal(path: &Path, tests: &mut Vec<TestPair>) {
```
Comment (nbolton) [inline src/test_pair.rs]: Perhaps we can come up with a better name than `find_all_tests_in_directory_internal`. When I see something like this, it makes me suspect some kind of design flaw.

**Review by nbolton — APPROVED** [review]
Unfortunately, as I'm so swamped, I don't have time to wrap my head around this change completely. Only one minor comment (at risk of being bike shed) so approving.

**Comment by nbolton** [comment]
@aquacash5 Up to you if you want to merge or improve the function name.

**Comment by nbolton** [comment]
@aquacash5 I edited the PR title. Would you say this is more accurate around the primary change?

**Pull request merged**